### DISCUSSION TITLE
Color modes switcher

### DISCUSSION THREAD

**Discussion by breatfr** [twbs/bootstrap#38848] [Q&A]
hi,

i already put a button to switch between dark and light mode with success even if i don't know js language but i have 2 questions:

1°) i would like to know if it's possible to save the user choice in cookie or other and how because actually if i reload the page the setting isn't saved.

here my codes:
the button:
`<a class="nav-link pointer" id="theme-switcher"><span class="fa fa-adjust fa-lg fa-fw" aria-hidden="true" title="Changer la couleur du site"></span></a>`

the js:
```
document.getElementById('theme-switcher').addEventListener('click', () => {
    if (document.documentElement.getAttribute('data-bs-theme') == 'dark') {
        document.documentElement.setAttribute('data-bs-theme', 'light')
    } else {
        document.documentElement.setAttribute('data-bs-theme', 'dark')
    }
})
```

2°) is it possible to add more color-scheme like blue, green, etc for exemple and if yes how ?

**Comment by julien-deramond** [comment]
Hey @breatfr 

> 1°) i would like to know if it's possible to save the user choice in cookie or other and how because actually if i reload the page the setting isn't saved.

We provide our own `color-modes.js` as an example which is used in the Bootstrap documentation: https://getbootstrap.com/docs/5.3/customize/color-modes/#javascript. In this file, you can see that we store the user choice in the local storage.

> 2°) is it possible to add more color-scheme like blue, green, etc for exemple and if yes how ?

You can create [custom color modes](https://getbootstrap.com/docs/5.3/customize/color-modes/#custom-color-modes).

**Reply by breatfr** [reply]
hi @julien-deramond and thanks for links.

i already tried this js but he don't work on my test [website](https://test.breat.fr/Bootstrap/fr/) but maybe it's normal because i use this `<html lang="fr" data-bs-theme="dark">` and not `<div data-bs-theme="dark"></div>`?

i keep the link to create custom color modes for later, before my priority is to fix the user choice 😄

**Reply by julien-deramond** [reply]
`<html lang="fr" data-bs-theme="dark">` is fine. If you change the value directly in the browser to `data-bs-theme="light"`, you'll see it works well. So you probably just need to tweak a bit our JS file to adapt it for your need. Basically, you need to save the user choice into the local storage and apply it by dynamically changing `data-bs-theme` value.

**Reply by breatfr** [reply]
hi @julien-deramond the problem is the js here https://getbootstrap.com/docs/5.3/customize/color-modes/#javascript is too hard for me to modify compared to my little code (I got it on internet too but can't remember where)

for exemple
```
document.getElementById('theme-switcher').addEventListener('click', () => {
    if (document.documentElement.getAttribute('data-bs-theme') == 'dark') {
        document.documentElement.setAttribute('data-bs-theme', 'light')
    } else {
        document.documentElement.setAttribute('data-bs-theme', 'dark')
    }
})
```
with this code i know easily even if i don't know js how to enable the js with a button, just need put one with an id="theme-switcher" like i show in my 1st post but with the https://getbootstrap.com/docs/5.3/customize/color-modes/#javascript I can't find how to enable him

**Reply by breatfr** [reply]
like i didn't arrive to make him works, i tried by copying the dropdown button who make this on [https://getbootstrap.com/](https://getbootstrap.com/), because i supposed (maybe bad) the website use this method, i replaced all bootstrap icons by the fork awesome ones, because i don't use bootstrap icons. the dropdown menu for choose is here but don't work too.

in my `<head>` i put the js code from [https://getbootstrap.com/docs/5.3/customize/color-modes/#javascript](https://getbootstrap.com/docs/5.3/customize/color-modes/#javascript) in .js file.

here my menu code for this function:
```
                <li class="nav-item dropdown">
                    <button class="btn btn-link nav-link py-2 px-0 px-lg-2 dropdown-toggle d-flex align-items-center" id="bd-theme" type="button" aria-expanded="false" data-bs-toggle="dropdown" data-bs-display="static" aria-label="Toggle theme (dark)">
                        <span class="fa fa-moon fa-lg fa-fw theme-icon-active" aria-hidden="true" title="Changer la couleur du site"></span>
                        <span class="d-lg-none ms-2" id="bd-theme-text">Toggle theme</span>
                    </button>
                    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="bd-theme-text">
                        <li>
                            <button type="button" class="dropdown-item d-flex align-items-center" data-bs-theme-value="auto" aria-pressed="false">
                            <span class="fa fa-adjust fa-fw me-2 opacity-50 theme-icon" aria-hidden="true"></span>
                            Auto
                            <span class="fa fa-check fa-fw ms-auto d-none" aria-hidden="true"></span>
                            </button>
                        </li>
                        <li>
                            <button type="button" class="dropdown-item d-flex align-items-center" data-bs-theme-value="light" aria-pressed="false">
                            <span class="fa fa-sun fa-fw me-2 opacity-50 theme-icon" aria-hidden="true"></span>
                            Clair
                            <span class="fa fa-check fa-fw ms-auto d-none" aria-hidden="true"></span>
                            </button>
                        </li>
                        <li>
                            <button type="button" class="dropdown-item d-flex align-items-center active" data-bs-theme-value="dark" aria-pressed="true">
                            <span class="fa fa-moon fa-fw me-2 opacity-50 theme-icon" aria-hidden="true"></span>
                            Sombre
                            <span class="fa fa-check fa-fw ms-auto d-none" aria-hidden="true"></span>
                            </button>
                        </li>
                    </ul>
                </li>
```

someone can help me ?

**edit**
maybe a stupid question, but do i need add jquery to make this js work ? i know some js need jquery so maybe it's that my problem because i didn't added it.
